pennylane/ops/qubit/time_dependent_hamiltonian.py

```diff
@@ -0,0 +1,218 @@
+    In many cases, Hamiltonians can be constructed using Pythonic arithmetic operations.
+    For example:
+
+    >>> qml.Hamiltonian([1.], [qml.PauliX(0)]) + 2 * qml.PauliZ(0) @ qml.PauliZ(1)
+
+    is equivalent to the following Hamiltonian:
+
+    >>> qml.Hamiltonian([1, 2], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliZ(1)])
+
+    Addition of TDHamiltonians with TDHamiltonians, Hamiltonians or other observables is possible with
+    tensor-valued coefficients, i.e.,
+
+    >>> H1 = qml.Hamiltonian(torch.tensor([1.]), [qml.PauliX(0)])
+    >>> H2 = qml.Hamiltonian(torch.tensor([2., 3.]), [qml.PauliY(0), qml.PauliX(1)])
+    >>> obs3 = [qml.PauliX(0), qml.PauliY(0), qml.PauliX(1)]
+    >>> H3 = qml.Hamiltonian(torch.tensor([1., 2., 3.]), obs3)
+    >>> H3.compare(H1 + H2)  # ToDo: compare function not implemented, modify documentation
+
+    # ToDo: do we want any grouping behaviour akin to options on Hamiltonian class?
+
+    """
+
+    num_wires = qml.operation.AnyWires
+    grad_method = "A"  # supports analytic gradients
+
+    def __init__(
+        self,
+        coeffs,
+        observables,
+        id=None,
+        do_queue=True,
+    ):
+
+        if qml.math.shape(coeffs)[0] != len(observables):
+            raise ValueError(
+                "Could not create valid Hamiltonian; "
+                "number of coefficients and operators does not match."
+            )
+
+        for obs in observables:
+            if not isinstance(obs, Observable):
+                raise ValueError(
+                    "Could not create circuits. Some or all observables are not valid."
+                )
+
+        self._ops = list(observables)
+        self._coeffs = coeffs
+
+        self.H_drift_coeffs = []
+        self.H_drift_ops = []
+        self.H_ts_fs = []
+        self.H_ts_ops = []
+
+        # ToDo: so far this only deals with scalar coefficients for H_drift - should it accept tensor_like?
+
+        for coeff, obs in zip(coeffs, observables):
+            if callable(coeff):
+                self.H_ts_fs.append(coeff)
+                self.H_ts_ops.append(obs)
+            else:
+                self.H_drift_coeffs.append(coeff)
+                self.H_drift_ops.append(obs)
+
+        if len(self.H_ts_ops) == 0:
+            raise RuntimeError(
+                "Found no time-dependent terms while initializing time dependent Hamiltonian "
+                f"with coefficients {self.coeffs}"
+            )
+
+        self.H_drift = self._H_drift
+        self.H_ts = self._H_ts
+
+        self._hyperparameters = {"ops": self._ops}
+
+        self._wires = qml.wires.Wires.all_wires([op.wires for op in self.ops], sort=True)
+
+        self.return_type = None
+
+        super().__init__(*self._coeffs, wires=self._wires, id=id, do_queue=do_queue)
+
+    def __call__(self, params, t, wire_order=None):
+        # TODO: jax-optimize this?
+        if wire_order is None:
+            wire_order = self.wires
+
+        if len(self.H_ts_ops) == 1:
```

Comment:
`if` clauses can cause problems when jitting, is this one here strictly necessary or can it be avoided? (It is the same code twice, but it is also just a draft so maybe there is more to come here?)

Reply:
Actually, it looks in the new UI document like this function shouldn't be returning a matrix anymore, but rather an Operator. That will not be jittable anyway, since it returns a PennyLane object. It will the the evaluation of the operator in ParametrizedEvolution that we want to jit.

Reply:
Also I missed Tom's comment that whether or not we want the a wires argument here isn't settled yet.

Reply:
It's ok for now to leave out the `wires` argument, we could add this later.